web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs

```diff
@@ -0,0 +1,345 @@
+                        archive_request: x.archive_request.into(),
+                        backend_requests: Default::default(), // This is not used, instead we modify the field later
+                        no_servers: 0.into(), // This is not relevant in the new version
+                        error_response: x.error_response.into(),
+                        response_bytes: int_response_bytes.into(),
+                        response_millis: int_response_millis.into(),
+                        // We just don't have this data
+                        response_from_backup_rpc: false.into(), // I think we did not record this back then // Default::default()
+                    };
+
+                    // (3) Send through a channel to a stat emitter
+                    // Send it to the stats sender
+                    if let Some(stat_sender_ref) = stat_sender.as_ref() {
+                        // info!("Method is: {:?}", x.clone().method);
+                        let mut response_stat = RpcQueryStats::new(
+                            x.clone().method,
+                            authorization.clone(),
+                            Arc::new(request_metadata),
+                            (int_response_bytes)
+                                .try_into()
+                                .context("sum bytes average is not calculated properly")?,
+                        );
+                        // Modify the timestamps ..
+                        response_stat.modify_struct(
+                            int_response_millis,
+                            x.period_datetime.timestamp(), // I suppose timestamp is millis as well ... should check this in the (prod) database
+                            int_backend_requests,
+                        );
+                        // info!("Sending stats: {:?}", response_stat);
+                        stat_sender_ref
+                            // .send(response_stat.into())
+                            .send_async(response_stat.into())
+                            .await
+                            .context("stat_sender sending response_stat")?;
+                        // info!("Send! {:?}", stat_sender);
+                    } else {
+                        panic!("Stat sender was not spawned!");
+                    }
+
+                    // Create a new stats object
+                    // Add it to the StatBuffer
+                }
+
+                // Let the stat_sender spawn / flush ...
+                // spawned_app.app.stat_sender.aggregate_and_save_loop()
+                // Send a signal to save ...
+            }
+
+            // (3) Await that all items are properly processed
+            // TODO: Await all the background handles
+
+            // Only after this mark all the items as processed / completed
+
+            // If the items are in rpc_v2, delete the initial items from the database
+
+            // return Ok(());
```

Comment:
the code after `// Finally also send a shutdown signal` should probably be here if you want to make sure the stat sender is done before updating the migrated column

Reply:
See above comment: anything above will make it respawn at each batch as I understand, see https://github.com/yenicelik/web3-proxy/blob/stats_v3/web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs#L274